### PR TITLE
Fixes #80 - Adding authentication support

### LINKED ISSUE #80: Enable Mesos Authentication Support

### PR DESCRIPTION
**Pull request by mcgin**

### CODE REVIEW

**Comment by keis** [comment]
Thanks for the PR @mcgin 

Please run gofmt to get the sorting of imports correct but otherwise this looks good to me.

File: scheduler/driver.go [hunk 1]
```diff
@@ -19,9 +24,45 @@ func getFrameworkID() *mesos.FrameworkID {
...
+func getCredential() (*mesos.Credential, error) {
+	if viper.IsSet("credential_file") {
+		content, err := ioutil.ReadFile(viper.GetString("credential_file"))
+		if err != nil {
+			logrus.WithError(err).WithFields(logrus.Fields{
+				"credential_file": viper.GetString("credential_file"),
+			}).Error("Unable to read credential_file")
+			return nil, err
+		}
+		fields := strings.Fields(string(content))
```
Comment (keis) [inline scheduler/driver.go]: hmm, we should check that we get exactly to fields out of this

File: scheduler/driver.go [hunk 2]
```diff
@@ -19,9 +24,45 @@ func getFrameworkID() *mesos.FrameworkID {
...
+		logrus.WithField("principal", fields[0]).Info("Successfully loaded principal")
+		return &mesos.Credential{
+			Principal: proto.String(fields[0]),
+			Secret:    proto.String(fields[1]),
+		}, nil
+	}
+	logrus.Info("No credentials specified in configuration")
```
Comment (alde) [inline scheduler/driver.go]: I'd probably prefer to have this as a debug message. Granted, it probably won't be very spammy, but I'd say it's not really relevant in a non-debug scenario.

**Comment by alde** [comment]
Thank you for the PR :) It looks quite good, but like @keis mentioned we should probably make sure the credential_file has a valid format too.

File: scheduler/driver.go (hunk 1 above)
Comment (mcgin) [inline scheduler/driver.go]: Makes sense have added the check

**Comment by mcgin** [comment]
PR updated with suggested changes.

**Comment by alde** [comment]
lgtm
looks good to me

**Comment by keis** [comment]
Merged and tagged as v0.17.0 :+1:

**Pull request merged**